test/suite/test_compat02.py

```diff
@@ -56,20 +56,27 @@ class test_compat02(wttest.WiredTigerTestCase, suite_subprocess):
     # The API uses only the major and minor numbers but accepts with
     # and without the patch number. Test one on release and the
     # required minimum just for testing of parsing.
-    #
+
     compat_create = [
         ('def', dict(create_rel='none', log_create=3)),
         ('32', dict(create_rel="3.2", log_create=3)),
+        ('31', dict(create_rel="3.1", log_create=3)),
         ('30', dict(create_rel="3.0", log_create=2)),
         ('26', dict(create_rel="2.6", log_create=1)),
```

Comment:
Do we still want to support compatibility testing for 2.6? It feels more sustainable to remove the oldest version while introducing a newer version into the tests, otherwise we're running more and more permutations of tests.

Reply:
As long as the code supports log version 1 compatibility then yes, we need to test a version that uses log version 1. We need to test moving up and down between them as long as the code allows that.

Reply:
@sueloverso, I've documented the process for updating the compatibility version for next release, Luke has given me feedback and I'll make some improvements to it. Could you please review this PR when you get a chance?

Reply:
Actually @lukech has pointed out a potential issue, so hold for the time being.

Reply:
In order to address @lukech  concerns I've updated the comment which explains how to add version to the compat max and compat min dictionaries. The tests as they are I think are correct. So this PR is ready to be reviewed again. Apologies for the confusion.